From f34eb45d0ed2de85c9f1789c7b0061f7d75c25bc Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
Date: Wed, 17 Feb 2016 16:58:29 +0100
Subject: [PATCH] drivers/gpio/Kconfig: enable GPIO configs for LS1021A SoC

This patch enable configs for GPIOs found on
Freescale's LS1021A SOCs

It also makes mutual exclusion with ARCH_MXC,
since the GPIO controller on LS1020A SOC
is different from ARCH_MXC GPIO controllers

Signed-off-by: Shaveta Leekha <shaveta@freescale.com>
Change-Id: Id5dd8947e2b0f41eb7ea9c3929ce5db8a245b007
Reviewed-on: http://git.am.freescale.net:8181/25120
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Jingchang Lu <jingchang.lu@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
[ported from fsl-sdk v1.9]
Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 drivers/gpio/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
index b18bea0..139dde0 100644
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -283,7 +283,7 @@ config GPIO_MPC5200
 config GPIO_MPC8XXX
 	bool "MPC512x/MPC8xxx GPIO support"
 	depends on PPC_MPC512x || PPC_MPC831x || PPC_MPC834x || PPC_MPC837x || \
-		   FSL_SOC_BOOKE || PPC_86xx
+		   FSL_SOC_BOOKE || PPC_86xx || SOC_LS1021A
 	help
 	  Say Y here if you're going to use hardware that connects to the
 	  MPC512x/831x/834x/837x/8572/8610 GPIOs.
@@ -296,7 +296,7 @@ config GPIO_MVEBU
 
 config GPIO_MXC
 	def_bool y
-	depends on ARCH_MXC
+	depends on ARCH_MXC && !SOC_LS1021A
 	select GPIO_GENERIC
 	select GENERIC_IRQ_CHIP
 
